llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -918,6 +918,16 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
     // value with the others blended into it.
 
     unsigned StartIndex = 0;
+    for (unsigned I = 0; I != Blend->getNumIncomingValues(); ++I) {
+      // If a value's mask is only used by the blend then is can be deadcoded.
+      // TODO: Find the most expensive mask that can be deadcoded.
```

Comment:
Perhaps another TODO is to find a mask that are used only by multiple blends, which can deadcoded if handled together, along with a testcase having two or more suitable original phi's.

Reply:
I've updated the comment.  I'll see about creating a test case separately as it's not specifically relevant to this PR.

Reply:
All phi's of a block are converted to blends with the same edge masks, so looking for a mask with a single blend using it restricts this patch to single phi'd blocks only.

Reply:
Thanks @ayalz, this was very useful.  I've added a multi-phi test to if-conversion-nest.ll to highlight the TODO.